src/server/jobs/update-bcast.js

```diff
@@ -0,0 +1,271 @@
+ */
+async function insertMediaEntries(client, session, entries) {
+  // logger.info(entries);
+  const result = await client
+    .db(process.env.DB_DATABASE)
+    .collection('bruincastmedia')
+    .insertMany(entries, { session });
+  logger.info(`Added ${result.insertedCount} record(s).`);
+}
+
+/**
+ * Deletes all media entries for a given term and Class ID
+ * @param  {MongoClient} client A connected MongoClient
+ * @param  {ClientSession} session A session started by client
+ * @param  {String} classTerm A class term formatted as YYQ
+ * @param  {String} classIDnum A 9-digit Class ID
+ */
+async function deleteMediaEntriesForClass(
+  client,
+  session,
+  classTerm,
+  classIDnum
+) {
+  const query = {
+    term: classTerm,
+    classID: classIDnum,
+  };
+  const result = await client
+    .db(process.env.DB_DATABASE)
+    .collection('bruincastmedia')
+    .deleteMany(query, { session });
+  logger.info(`Deleted ${result.deletedCount} record(s).`);
+}
+
+/**
+ * For a given term and Class ID, delete all records and insert new entries
+ * @param  {MongoClient} client A connected MongoClient
+ * @param  {String} classTerm A class term formatted as YYQ
+ * @param  {String} classIDnum A 9-digit Class ID
+ * @param  {[Object]} entries An array of JSON-formatted objects for media entries
+ */
+async function updateRecordsForClass(client, classTerm, classIDnum, entries) {
+  const session = client.startSession();
+  session.startTransaction();
+  try {
+    await deleteMediaEntriesForClass(client, session, classTerm, classIDnum);
+    await insertMediaEntries(client, session, entries);
+    await session.commitTransaction();
+    session.endSession();
+  } catch (err) {
+    await session.abortTransaction();
+    session.endSession();
+    throw err;
+  }
+}
+
+async function main() {
+  // Log in to BruinCast API and store cookie
+  await loginBruinCast();
+
+  const dbURL = `${process.env.DB_URL}${process.env.DB_DATABASE}?replicaSet=${process.env.DB_REPLSET}`;
+  const client = new MongoClient(dbURL, { useUnifiedTopology: true });
+
+  try {
+    await client.connect();
+    logger.info(`Connected database at ${dbURL}`);
+
+    const currentTerm = process.argv[2];
+    if (!currentTerm) {
+      throw new Error(
+        'Missing command line argument: Term required. (e.g. 20S)'
+      );
+    }
+
+    const formattedTerm = convertTerm(currentTerm);
+    let numTotalRecords = 0;
+
+    logger.info(`Updating records for term: ${currentTerm}`);
+
+    // From the API, get Class IDs with available media for the given term
+    await getCourses(formattedTerm, async function(coursesResponse) {
+      const courses = JSON.parse(coursesResponse);
+
+      // For each term and Class ID, get the associated media and update records
+      for await (const course of courses) {
+        const currentClassID = course['srs #'];
+        logger.info(`Updating record(s) for Class ID: ${currentClassID}`);
```

Comment:
@chern When you get the ability to query for shortname, make sure ti also log the shortname found for the given Class ID.